src/libcollections/vec_deque.rs

```diff
@@ -1538,16 +1535,17 @@ impl<'a, T> Iterator for IterMut<'a, T> {
             return None;
         }
         let tail = self.tail;
-        self.tail = wrap_index(self.tail + 1, self.cap);
+        self.tail = wrap_index(self.tail + 1, self.ring.len());
 
         unsafe {
-            Some(&mut *self.ptr.offset(tail as isize))
+            let elem = self.ring.get_unchecked_mut(tail);
+            Some(mem::transmute(elem))
```

Comment:
I think we tend to prefer avoiding `transmute` wherever possible, could this be written as `Some(&mut *(elem as *mut _))`?